evalml/pipelines/classification_pipeline.py

```diff
@@ -50,9 +51,20 @@ def fit(self, X, y):
 
         Returns:
             self
+
+        Raises:
+            ValueError: If the number of unique classes in y are not appropriate for the type of pipeline.
         """
         X = infer_feature_types(X)
         y = infer_feature_types(y)
+
+        if self.problem_type == ProblemTypes.BINARY and y.nunique() != 2:
+            raise ValueError(f"Binary pipelines require y to have 2 unique classes!")
+        elif self.problem_type == ProblemTypes.MULTICLASS and y.nunique() in [1, 2]:
+            raise ValueError(
+                "Multiclass pipelines require y to have 3 or more unique classes!"
+            )
```

Comment:
I think I asked this on your previous PR, but is there anyway we can tie this to the logic in `invalid_target_data_check`?  That data check, albeit somewhat complex, does check for this.  But at the same time, this is starting to feel weird, adding bits and pieces of certain data checks back into the actual pipelines.  @angela97lin , any thoughts?  Are we moving in a direction contrary to the philosophy we had in mind for data checks?

Reply:
I don't think that's a bad idea at all @chukarsten, especially since we already have a check for whether or not a target has enough instances of a class in a classification problem--it seems to fit the philosophy for data checks quite well.

Pretty interesting problem, maybe this is worth a discussion with the larger group? I wonder if we need to better draw the line between when we want to throw an exception to strongly disallow something vs use data checks.

Reply:
Calling attention to this issue a filed forever ago that I think tackles the same problem https://github.com/alteryx/evalml/issues/968

Basically, let pipelines accept data checks on init so that we can run data checks prior to fit. 

I think that's just one particular solution to the problem at hand but I think it's worth considering.